library/src/main/java/com/pokegoapi/api/map/Map.java

```diff
@@ -584,90 +586,30 @@ public FortSearchResponse searchFort(FortData fortData) throws LoginFailedExcept
 				.setPlayerLatitude(api.getLatitude())
 				.setPlayerLongitude(api.getLongitude())
 				.build();
-		ServerRequest serverRequest = new ServerRequest(RequestType.FORT_SEARCH, reqMsg);
 
-		api.getRequestHandler().sendServerRequests(serverRequest);
 
-		FortSearchResponse response;
-		try {
-			response = FortSearchResponse.parseFrom(serverRequest.getData());
-		} catch (InvalidProtocolBufferException e) {
-			throw new RemoteServerException(e);
-		}
-		return response;
-	}
+		AsyncServerRequest serverRequest = new AsyncServerRequest(RequestType.LEVEL_UP_REWARDS, reqMsg, api);
+		return AsyncHelper.toBlocking(
+				api.getRequestHandler().sendAsyncServerRequests(serverRequest).map(new Func1<ByteString, FortSearchResponse>() {
 
-	/**
-	 * Encounter pokemon encounter response.
-	 *
-	 * @param catchablePokemon the catchable pokemon
-	 * @return the encounter response
-	 * @throws LoginFailedException  the login failed exception
-	 * @throws RemoteServerException the remote server exception
-	 */
-	@Deprecated
-	public EncounterResponse encounterPokemon(MapPokemon catchablePokemon)
-			throws LoginFailedException, RemoteServerException {
+					@Override
+					public FortSearchResponse call(ByteString bytes) {
+						FortSearchResponse response;
+						try {
+							response = FortSearchResponse.parseFrom(bytes);
+						} catch (InvalidProtocolBufferException e) {
+							throw new AsyncRemoteServerException(e);
+						}
+
+						return response;
+					}
+				})
+		);
 
-		EncounterMessageOuterClass.EncounterMessage reqMsg = EncounterMessageOuterClass.EncounterMessage.newBuilder()
-				.setEncounterId(catchablePokemon.getEncounterId())
-				.setPlayerLatitude(api.getLatitude())
-				.setPlayerLongitude(api.getLongitude())
-				.setSpawnPointId(catchablePokemon.getSpawnPointId())
-				.build();
-		ServerRequest serverRequest = new ServerRequest(RequestType.ENCOUNTER, reqMsg);
-		api.getRequestHandler().sendServerRequests(serverRequest);
-
-		EncounterResponse response;
-		try {
-			response = EncounterResponse.parseFrom(serverRequest.getData());
-		} catch (InvalidProtocolBufferException e) {
-			throw new RemoteServerException(e);
-		}
-		return response;
 	}
 
-	/**
-	 * Catch pokemon catch pokemon response.
-	 *
-	 * @param catchablePokemon      the catchable pokemon
-	 * @param normalizedHitPosition the normalized hit position
-	 * @param normalizedReticleSize the normalized reticle size
-	 * @param spinModifier          the spin modifier
-	 * @param pokeball              the pokeball
-	 * @return the catch pokemon response
-	 * @throws LoginFailedException  the login failed exception
-	 * @throws RemoteServerException the remote server exception
-	 */
-	@Deprecated
-	public CatchPokemonResponse catchPokemon(
```

Comment:
synchronous method = bye bye :)
methods that are handled on commonRequest = bye bye

also, all the lists that will throw modification exceptions have to go, same for items that have duplicate stuff on them, have to be converted on map. clear method on inventory will go and inventory need to be updated in a proper way, not cleaning and populating it every time